SendGrid/SendGrid/Client.cs

```diff
@@ -0,0 +1,123 @@
+        /// <param name="baseUri">Base SendGrid API Uri</param>
+        public Client(string apiKey, string baseUri = "https://api.sendgrid.com/")
+        {
+            _baseUri = new Uri(baseUri);
+            _apiKey = apiKey;
+            ApiKeys = new APIKeys(this);
+        }
+
+        /// <summary>
+        ///     Create a client that connects to the SendGrid Web API
+        /// </summary>
+        /// <param name="method">HTTP verb, case-insensitive</param>
+        /// <param name="endpoint">Resource endpoint, do not prepend slash</param>
+        /// <param name="data">An object representing the resource's data</param>
+        /// <returns>An asyncronous task</returns>
+        private async Task<HttpResponseMessage> RequestAsync(string method, string endpoint, Object data)
+        {
+            using (var client = new HttpClient())
+            {
+                try
+                {
+                    client.BaseAddress = _baseUri;
+                    client.DefaultRequestHeaders.Accept.Clear();
+                    client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+                    client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", _apiKey);
+                    var version = Assembly.GetExecutingAssembly().GetName().Version.ToString();
+                    client.DefaultRequestHeaders.TryAddWithoutValidation("User-Agent", "sendgrid/" + version + ";csharp");
+
+                    switch (method.ToLower())
+                    {
+                        case "get":
+                            return await client.GetAsync(endpoint);
+                        case "post":
+                            return await client.PostAsJsonAsync(endpoint, data);
+                        case "patch":
+                            endpoint = _baseUri + endpoint;
+                            StringContent content = new StringContent(data.ToString(), Encoding.UTF8, "application/json");
+                            HttpRequestMessage request = new HttpRequestMessage
+                            {
+                                Method = new HttpMethod("PATCH"),
+                                RequestUri = new Uri(endpoint),
+                                Content = content
+                            };
+                            return await client.SendAsync(request);
+                        case "delete":
+                            return await client.DeleteAsync(endpoint);
+                        default:
+                            HttpResponseMessage response = new HttpResponseMessage();
+                            response.StatusCode = HttpStatusCode.MethodNotAllowed;
+                            response.Content = new StringContent("Bad method call: " + method);
+                            return response;
+                    }
+                }
+                catch (HttpRequestException hre)
```

Comment:
In the event of an exception do you want to return the runtime specific .net exception to the caller?  Perhaps returning a more user-friendly message would be better.